index.js

```diff
@@ -245,6 +245,41 @@ Buffer.allocUnsafeSlow = function (size) {
   return allocUnsafe(size)
 }
 
+/**
+ * Copies the underlying memory of `view` into a new `Buffer`.
+ */
+Buffer.copyBytesFrom = function copyBytesFrom (view, offset, length) {
+  if (!ArrayBuffer.isView(view) || !view.subarray) {
+    throw new errors.ERR_INVALID_ARG_TYPE('view', 'TypedArray', view)
+  }
+
+  if (view.length === 0) return createBuffer(0)
+
+  if (offset !== undefined || length !== undefined) {
+    if (offset !== undefined) {
+      validateInteger(offset, 'offset')
+      if (offset >= view.length) return createBuffer(0)
+    } else {
+      offset = 0
+    }
+
+    let end
+
+    if (length !== undefined) {
+      validateInteger(length, 'length')
+      end = offset + length
+    } else {
+      end = view.length
+    }
+
+    view = view.subarray(offset, end)
+  }
+
+  view = new Uint8Array(view.buffer, view.byteOffset, view.byteLength)
+
+  return fromArrayView(view)
```

Comment:
```suggestion
  if (offset !== undefined || length !== undefined) {
    if (offset !== undefined) validateInteger(offset, 'offset')
    if (length !== undefined) validateInteger(length, 'length')
    const offset_ = offset ?? 0
    const end = (offset + length) ?? view.length
    view = view.subarray(offset, end)
  }
  
  view = new Uint8Array(view.buffer, view.byteOffset, view.byteLength)
  return fromArrayView(view)
```

Reply:
Generally, a preference for flat/functional style, even if the actual number of branches is the same

Reply:
This code was ported from the node.js code as faithfully as possible to ensure accurate behavior. I can change it though.

While we're on the subject: what is the minimum ES version we support? Since the `??` operator is an ES2020 feature, it would be good to have clarity on that. BigInt is also an ES2020 feature, but we include graceful degradation for the bigint methods.